triggers/daml/Daml/Trigger.daml

```diff
@@ -98,10 +110,12 @@ dedupCreate t = do
   -- map to make these lookups cheaper.
   let cmds = concat $ map snd (Map.toList aState.commandsInFlight) <> map commands aState.emittedCommands
   unless (any ((Some t ==) . fromCreate) cmds) $
-    void $ emitCommands [createCmd t]
+    void $ emitCommands [createCmd t] []
 
 -- | Exercise the choice on the given contract if it is not already
 -- in flight.
+-- dedupExercise will also mark the contract id as pending so it will not show up
+-- in `getContracts` in the next run.
```

Comment:
I'm not sure I understand the motivation behind this. Is this to model consuming choices? If so, does this interact with nonconsuming choices?

Reply:
Good point, I’ve removed the automatic pending thing here and then it behaves like `exerciseByKeyCmd`.